dmcontent/markdown.py

```diff
@@ -0,0 +1,44 @@
+from typing import Mapping
+from xml.etree.ElementTree import Element
+
+from markdown.extensions import Extension
+from markdown.treeprocessors import Treeprocessor
+
+
+class _ClassAddingTreeprocessor(Treeprocessor):
+    _mapping: Mapping[str, str]
+
+    def __init__(self, mapping: Mapping[str, str]):
+        self._mapping = mapping
+
+    def run(self, root: Element):
+        self._run(root)
+
+    def _run(self, element: Element):
+        # recurse
+        for child in element:
+            self._run(child)
+
+        # mutate each affected element in-place
+        if element.tag in self._mapping:
+            orig_class = element.attrib.get("class", "")
+            element.set("class", (orig_class + " " if orig_class else "") + self._mapping[element.tag])
+
+
+class GovUkFrontendExtension(Extension):
```

Comment:
This should probably be named `GOVUKFrontendExtension` to be consistent with [govuk-frontend](https://github.com/alphagov/govuk-frontend/blob/899264cd89559d31f0a790682254d94ecf66418b/package/govuk/common.js)

Reply:
:+1: